### PR TITLE
systemd hardening

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: module.nix [hunk 1]
```diff
@@ -57,23 +64,29 @@ in
...
       };
     };
-
-    users = {
-      users.harmonia = {
-        isSystemUser = true;
-        group = "harmonia";
-      };
-      groups.harmonia = { };
-    };
```
Comment (mikepurvis) [inline module.nix:77]: This broke some logic I inherited from @NinjaTrappeur that was passing a signing key into Harmonia from a sops location:

```
      nix-signing-key = {
        restartUnits = [ "harmonia.service" ];
        owner = config.users.users.harmonia.name;
        mode = "0440";
      };
```

Looks like the two ways forward here are systemd's `LoadCredential` scheme (which will make a file available read-only to the unit's user/environment) or else switching from a file to an environment variable. Any thoughts on which of these would be preferable or if there's another option I'm missing?

Comment (mikepurvis) [reply module.nix:77]: Ah, there's an existing discussion on this in https://github.com/Mic92/sops-nix/issues/198.

Comment (Mic92) [reply module.nix:77]: LoadCredential that is used, shouldn't require a user. Here an example:

https://github.com/Mic92/dotfiles/blob/e8829eed755bf154579668033c962c962ef6a943/nixos/eve/modules/harmonia.nix#L3

Comment (mikepurvis) [reply module.nix:77]: Thanks! Yes, that's helpful and I do now see how the pieces fit together. I think our original deployment predated the introduction of LoadCredential on your side.

**Pull request merged**